src/main/java/uk/gov/companieshouse/ocr/api/image/extracttext/ImageOcrService.java

```diff
@@ -0,0 +1,137 @@
+
+            LOG.debugContext(textConversionResult.getResponseId(), "Number of pages to process " + totalPages, null);
+
+            StringBuilder documentText = new StringBuilder();
+            for (int currentPage = 0; currentPage < totalPages; currentPage++) {
+
+                textConversionResult.addPage();
+
+                LOG.infoContext(textConversionResult.getResponseId(), "Processed " + (currentPage * 100) / totalPages + "%", null);
+
+                documentText.append(extractTextOnCurrentPage(reader, textConversionResult, api, handle, currentPage));
+            }
+
+            textConversionResult.completeSuccess(documentText.toString());
+
+            LOG.infoContext(textConversionResult.getResponseId(),"Document MetaData", textConversionResult.metaDataMap());
+
+        } catch (IOException ex) {
+            var tce = new TextConversionException(textConversionResult.getResponseId(), ex);
+            throw tce;
+        } finally {
+            api.TessBaseAPIDelete(handle);
+        }
+    }
+
+    private String extractTextOnCurrentPage(ImageReader reader, TextConversionResult textConversionResult, TessAPI api,
+            TessBaseAPI handle, int currentPage) throws IOException {
+
+        var image = reader.read(currentPage);
+        var buf = ImageIOHelper.convertImageData(image);
+        var bpp = image.getColorModel().getPixelSize();
+        var bytespp = bpp / 8;
+        var bytespl = (int) Math.ceil(image.getWidth() * bpp / 8.0);
+
+        api.TessBaseAPISetImage(handle, buf, image.getWidth(), image.getHeight(), bytespp, bytespl);
+        var utf8TextPtr = api.TessBaseAPIGetUTF8Text(handle);
+        var pageText = utf8TextPtr.getString(0);
+        api.TessDeleteText(utf8TextPtr);
+
+        var ri = api.TessBaseAPIGetIterator(handle);
+        var level = TessPageIteratorLevel.RIL_TEXTLINE;
+  
+        if (ri != null) {
+            do {
+                var symbol = api.TessResultIteratorGetUTF8Text(ri, level);
+                var confidence = api.TessResultIteratorConfidence(ri, level);
+                if (symbol != null) {
+                    textConversionResult.addConfidence(confidence);
+                    LOG.trace("Confidence is " + confidence + " for " + symbol.getString(0));
+                } else {
+                    LOG.debug("Confidence is " + confidence + " for blank line" );
+                }
+            } while (api.TessResultIteratorNext(ri, level) == ITessAPI.TRUE);
+        }
+        return pageText;
+    }
+
+    private ImageReader createImageReader(ImageInputStream is) throws IOException {
+        if (is == null || is.length() == 0) {
+            throw new IOException("Invalid input stream");
+        }
+
+        Iterator<ImageReader> iterator = ImageIO.getImageReaders(is);
+        if (iterator == null || !iterator.hasNext()) {
+            throw new IOException("Image file format not supported by ImageIO: ");
```

Comment:
Is the file image format being unsupported the only way the iterator might be empty? Would "No file readers found for image content" be a better wording?

Reply:
Updated